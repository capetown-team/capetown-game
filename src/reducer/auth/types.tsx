export const AUTH_REQUEST = 'AUTH_REQUEST';
export const AUTH_SUCCESS = 'AUTH_SUCCESS';
export const AUTH_CHECK_FAILURE = 'AUTH_CHECK_FAILURE';

export const LOGOUT = 'LOGOUT';
export const LOGOUT_FAILURE = 'USER_FAILURE';
