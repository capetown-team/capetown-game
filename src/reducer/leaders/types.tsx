export const LOAD_LEADERBOARD = 'LOAD_LEADERBOARD';
export const LEADERBOARD_REQUEST = 'LEADERBOARD_REQUEST';
export const LEADERBOARD_ERROR = 'LEADERBOARD_ERROR';
export const LEADERBOARD_SUCCESS = 'LEADERBOARD_SUCCESS';
