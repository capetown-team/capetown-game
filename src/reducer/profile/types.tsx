export const CHANGE_PROFILE_VIEW = 'CHANGE_PROFILE_VIEW';
export const CHANGE_PROFILE = 'CHANGE_PROFILE';
