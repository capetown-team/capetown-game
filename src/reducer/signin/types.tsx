export const SIGNIN_REQUEST = 'SIGININ_REQUEST';
export const SIGNIN_SUCCESS = 'SIGININ_SUCCESS';
export const SIGNIN_FAILURE = 'SIGININ_CHECK_FAILURE';
