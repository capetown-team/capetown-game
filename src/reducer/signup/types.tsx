export const SIGNUP_REQUEST = 'SINGUP_REQUEST';
export const SIGNUP_SUCCESS = 'SINGUP_SUCCESS';
export const SIGNUP_FAILURE = 'SINGUP_FAILURE';
