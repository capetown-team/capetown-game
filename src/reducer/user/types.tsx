export const AUTH_REQUEST = 'AUTH_REQUEST';
export const AUTH_SUCCESS = 'AUTH_SUCCESS';
export const AUTH_CHECK_FAILURE = 'AUTH_CHECK_FAILURE';
export const AUTH_FAILURE = 'AUTH_FAILURE';

export const LOGOUT = 'LOGOUT';
export const USER_FAILURE = 'USER_FAILURE';
