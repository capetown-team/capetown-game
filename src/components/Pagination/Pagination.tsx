import React, { memo, FC, useMemo } from 'react';
import block from 'bem-cn-lite';
import { PaginateType } from '@/types.d';

import './Pagination.scss';

const b = block('pagination');

export type Props = {
  usersPerPage: number;
  totalUsers: number;
  currentPage: number;
  paginate: PaginateType;
  search?: string;
};

const Pagination: FC<Props> = ({
  usersPerPage,
  totalUsers,
  currentPage,
  paginate
}) => {
  const pageNumbers = useMemo(() => {
    const res: { num: number; current: boolean }[] = [];

    for (let i = 1; i <= Math.ceil(totalUsers / usersPerPage); i += 1) {
      if (i === currentPage) {
        res.push({ num: i, current: true });
      } else {
        res.push({ num: i, current: false });
      }
    }

    return res;
  }, [currentPage, totalUsers, usersPerPage]);

  return (
    <ul className={b()}>
      {pageNumbers.map(({ num, current }) => (
        <li key={num} className={b('list')}>
          <span
            onClick={() => paginate(num)}
            className={b('link', { active: !!current })}
            role="presentation"
          >
            {num}
          </span>
        </li>
      ))}
    </ul>
  );
};

const WrappedPagination = memo(Pagination);

export { WrappedPagination as Pagination };
