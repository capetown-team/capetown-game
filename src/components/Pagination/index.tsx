import Pagination from '@/components/Pagination/Pagination';

export default Pagination;
