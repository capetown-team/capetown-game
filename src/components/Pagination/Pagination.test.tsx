import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { PaginateType } from '@/types.d';
import { Pagination } from './Pagination';

describe('<Pagination />', () => {
  it('should renders correct <Button />', () => {
    const number = 1;
    const handlerPaginate: PaginateType = () => {
      return number + 2;
    };
    const wrapper = shallow(
      <Pagination
        usersPerPage={7}
        totalUsers={20}
        paginate={handlerPaginate}
        currentPage={1}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
