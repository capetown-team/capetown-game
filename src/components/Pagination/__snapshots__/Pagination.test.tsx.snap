// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should renders correct <Button /> 1`] = `
<ul
  className="pagination"
>
  <li
    className="pagination__list"
    key="1"
  >
    <span
      className="pagination__link pagination__link_active"
      onClick={[Function]}
      role="presentation"
    >
      1
    </span>
  </li>
  <li
    className="pagination__list"
    key="2"
  >
    <span
      className="pagination__link"
      onClick={[Function]}
      role="presentation"
    >
      2
    </span>
  </li>
  <li
    className="pagination__list"
    key="3"
  >
    <span
      className="pagination__link"
      onClick={[Function]}
      role="presentation"
    >
      3
    </span>
  </li>
</ul>
`;
